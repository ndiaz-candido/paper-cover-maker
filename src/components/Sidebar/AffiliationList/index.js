import AffiliationList from './AffiliationList';

export default AffiliationList;
