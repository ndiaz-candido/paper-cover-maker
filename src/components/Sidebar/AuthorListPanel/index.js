import AuthorListPanel from './AuthorListPanel';

export default AuthorListPanel;
