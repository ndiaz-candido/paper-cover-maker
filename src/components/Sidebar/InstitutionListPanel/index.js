import InstitutionListPanel from './InstitutionListPanel';

export default InstitutionListPanel;
