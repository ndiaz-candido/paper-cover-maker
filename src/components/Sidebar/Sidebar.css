@import "../../common/Colors.css";

:local {
  .sidebar {
    width: 100%;

    .affiliations {
      .lists {
        display: flex;

        .list {
          flex: 1 1 50%;
          padding: 0 15px;
        }
      }
    }
  }
}
