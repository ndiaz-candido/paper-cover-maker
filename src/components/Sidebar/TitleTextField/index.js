import TitleTextField from './TitleTextField';

export default TitleTextField;
