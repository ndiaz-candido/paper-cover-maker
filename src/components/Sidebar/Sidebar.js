import React from 'react';
import PropTypes from 'prop-types';
import AuthorListPanel from './AuthorListPanel';
import InstitutionListPanel from './InstitutionListPanel';
import TitleTextField from './TitleTextField';
import styles from './Sidebar.css';
import Header from './Header';
import Footer from './Footer';

const Sidebar = ({ onSave }) => (
  <div className={styles.sidebar}>
    <TitleTextField />
    <div className={styles.affiliations}>
      <Header title="Affiliations" />
      <div className={styles.lists}>
        <div className={styles.list}>
          <AuthorListPanel />
        </div>
        <div className={styles.list}>
          <InstitutionListPanel />
        </div>
      </div>
    </div>
    <Footer onSave={onSave} />
  </div>
);

Sidebar.propTypes = {
  onSave: PropTypes.func
};

Sidebar.defaultProps = {
  onSave: () => {}
};

export default Sidebar;
