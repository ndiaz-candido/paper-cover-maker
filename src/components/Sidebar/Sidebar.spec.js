import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar';

describe('Sidebar', () => {
  test('should render its default content', () => {
    const SidebarComponent = shallow(<Sidebar />);

    expect(SidebarComponent).toMatchSnapshot();
  });

  test('should call onSave function when the save button in the Footer is clicked', () => {
    const onSaveFn = jest.fn();
    const SidebarComponent = shallow(<Sidebar onSave={onSaveFn} />);

    SidebarComponent.find('Footer').prop('onSave')();

    expect(onSaveFn).toHaveBeenCalled();
  });
});
