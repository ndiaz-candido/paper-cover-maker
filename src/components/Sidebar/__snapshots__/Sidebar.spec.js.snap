// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render its default content 1`] = `
<div
  className="sidebar"
>
  <TitleTextArea />
  <div
    className="affiliations"
  >
    <Header
      actionsRenderer={[Function]}
      title="Affiliations"
    />
    <div
      className="lists"
    >
      <div
        className="list"
      >
        <AuthorListPanel />
      </div>
      <div
        className="list"
      >
        <InstitutionListPanel />
      </div>
    </div>
  </div>
  <Footer
    onSave={[Function]}
  />
</div>
`;
